pkg/icingadb/ha.go

```diff
@@ -274,15 +287,25 @@ func (h *HA) realize(ctx context.Context, s *icingaredisv1.IcingaStatus, t *type
 			switch errQuery {
 			case nil:
 				otherResponsible = true
-				if shouldLog {
+				if shouldLogRoutineEvents {
 					h.logger.Infow("Another instance is active",
 						zap.String("instance_id", instance.Id.String()),
 						zap.String("environment", envId.String()),
-						"heartbeat", instance.Heartbeat,
+						zap.Time("heartbeat", instance.Heartbeat.Time()),
 						zap.Duration("heartbeat_age", time.Since(instance.Heartbeat.Time())))
 				}
 			case sql.ErrNoRows:
-				takeover = true
+				fields := []any{
+					zap.String("instance_id", instance.Id.String()),
+					zap.String("environment", envId.String()),
+					zap.Duration("heartbeat_timeout", timeout)}
+
+				if !h.responsible {
+					h.logger.Infow("Preparing to take over HA as no instance is active", fields...)
+				} else if h.responsible && shouldLogRoutineEvents {
+					h.logger.Infow("Continuing being the active instance", fields...)
```

Comment:
Is that even possible? I can't imagine a situation in which we are preparing for the takeover but at the same time being responsible/the active instance.

Reply:
This condition will always be true if this Icinga DB instance is the active one.

The SQL query above selects all responsible instances which are not the current one.  Thus, `sql.ErrNoRows` will be returned.
https://github.com/Icinga/icingadb/blob/7d25da7815de6035e958c1bd0331f7669f95b090/pkg/icingadb/ha.go#L279-L285

That's how it looks on my current testing setup:
```
icingadb-master2  | 2024-03-14T11:25:02.914Z    INFO    high-availability       Continuing being the active instance    {"environment": "cef9c044b80d8cf516b031c0162ef41394b36498", "heartbeat_timeout": "1m0s"}
icingadb-master2  | 2024-03-14T11:26:02.913Z    INFO    high-availability       Continuing being the active instance    {"environment": "cef9c044b80d8cf516b031c0162ef41394b36498", "heartbeat_timeout": "1m0s"}
icingadb-master2  | 2024-03-14T11:27:02.914Z    INFO    high-availability       Continuing being the active instance    {"environment": "cef9c044b80d8cf516b031c0162ef41394b36498", "heartbeat_timeout": "1m0s"}
icingadb-master2  | 2024-03-14T11:28:02.913Z    INFO    high-availability       Continuing being the active instance    {"environment": "cef9c044b80d8cf516b031c0162ef41394b36498", "heartbeat_timeout": "1m0s"}
icingadb-master2  | 2024-03-14T11:29:02.915Z    INFO    high-availability       Continuing being the active instance    {"environment": "cef9c044b80d8cf516b031c0162ef41394b36498", "heartbeat_timeout": "1m0s"}
```

Reply:
Btw, by guarding this with the `shouldLogRoutineEvents` variable, this will be logged once every minute, similar to the opposite _"Another instance is active"_ message on the other node.

Reply:
I moved both down to `DEBUG`.

Reply:
I think the idea for the "another instance is active" message to be repeated every now and then was to be some kind of heartbeat thing saying "I'm fine, it's intentional that not much is happening here". Do we still need this? Not entirely sure without trying how this behaves now.

However, if this stays at debug level, the whole `shouldLogRoutineEvents` thing can be removed. The only point of this is so that the regular log isn't spammed with these messages.

Reply:
Valid point. I removed the `shouldLog` (or as I renamed it as `shouldLogRoutineEvents`) logic.